bulbs/contributions/signals.py

```diff
@@ -27,36 +27,55 @@ def call_update_role_rates(sender, instance, * args, **kwargs):
 def update_contributions(sender, instance, action, model, pk_set, **kwargs):
     """Creates a contribution for each author added to an article.
     """
-    # action = kwargs.get('action')
-    # import pdb; pdb.set_trace()
     if action != 'pre_add':
         return
     else:
-        # model = kwargs.get('model')
-        # pk_set = kwargs.get('pk_set')
         for author in model.objects.filter(pk__in=pk_set):
             update_content_contributions(instance, author)
 
 
-@receiver(post_save, sender=Content)
-def index_reportcontent(sender, instance, **kwargs):
+@receiver(post_save, sender=Contribution)
+def index_relations(sender, instance, **kwargs):
     """
-    Indexes the reporting document for the piece of content
+    We need to update the ReportContent object whenver a contribution is added.
     """
     try:
-        proxy = ReportContent.reference.get(id=instance.id)
+        proxy = ReportContent.reference.get(id=instance.content_id)
         proxy.index()
     except:
```

Comment:
I'd catch a specific exception here (NotFound) to avoid hiding unexpected errors